src/test/kotlin/lotto/domain/LottoTest.kt

```diff
@@ -0,0 +1,48 @@
+package lotto.domain
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class LottoTest {
+    @Test
+    fun `Illegal Lotto`() {
+        assertThrows<IllegalArgumentException> {
+            Lotto()
+        }
+    }
+
+    @Test
+    fun `Lotto has 6 numbers`() {
+        val ticket = listOf(1, 2, 3, 4, 5, 6)
+        assertThat(Lotto(ticket))
+    }
+
+    @Test
+    fun `Lotto has 6 not unique numbers`() {
+        val ticket = listOf(1, 1, 2, 3, 4, 5)
+        assertThrows<IllegalArgumentException> {
+            Lotto(ticket)
+        }
+    }
+
+    @Test
+    fun `Lotto numbers are in range`() {
+        val ticket = listOf(1, 2, 3, 4, 5, 45)
+        assertThat(Lotto(ticket))
+    }
+
+    @Test
+    fun `Lotto numbers are not in range`() {
+        val ticket = listOf(1, 100, 2, 3, 4, 5)
+        assertThrows<IllegalArgumentException> {
+            Lotto(ticket)
+        }
+    }
```

Comment:
The second test about lotto numbers not being in range could be improved by using boundary values such as `0` and `46`.